test/unit/winston/logger.test.js

```diff
@@ -779,150 +812,675 @@ describe('Logger Instance', function () {
+      const expectedOutput = [
+        {level: "info", message: "test message", someMap: new Map([["val1","c"], ["val2", "b"]])}
+      ];
+
+      const logger1 = winston.createLogger({
+        transports: [mockTransports.inMemory(levelOutput)]
+      });
+      const logger2 = winston.createLogger({
+        transports: [mockTransports.inMemory(logOutput)]
+      });
+      const logMeta = {
+        someMap: new Map([["val1","a"], ["val2", "b"], ["val1","c"]])
+      };
+
+      logger1.info("test message", logMeta);
+      logger2.info({level: "info", message: "test message", ...logMeta});
+
+      assume(levelOutput).eqls(logOutput);
+      assume(expectedOutput).eqls(levelOutput);
+      assume(expectedOutput).eqls(logOutput);
+    });
+
+    it('should support a Set', function() {
+      const expectedOutput = [
+        {level: "info", message: "test message", someSet: new Set(["a", "b"])}
+      ]
+
+      const logger1 = winston.createLogger({
+        transports: [mockTransports.inMemory(levelOutput)]
+      });
+      const logger2 = winston.createLogger({
+        transports: [mockTransports.inMemory(logOutput)]
+      });
+      const logMeta = {
+        someSet: new Set(["a","b", "a"])
+      };
+
+      logger1.info("test message", logMeta);
+      logger2.info({level: "info", message: "test message", ...logMeta});
+
+      assume(levelOutput).eqls(logOutput);
+      assume(expectedOutput).eqls(levelOutput);
+      assume(expectedOutput).eqls(logOutput);
+    });
+  });
+
+  describe('Metadata Precedence', () => {
+    describe('Single logger instance', () => {
+      it('should log to passed array correctly when using the `inMemory` transport', () => {
+        const expectedOutput = [
+          {message: "some message", level: "info"},
+          {message: "I'm a test", level: "warn"}
+        ];
+
+        const logger1 = winston.createLogger({
+          transports: [mockTransports.inMemory(levelOutput)]
+        });
+        const logger2 = winston.createLogger({
+          transports: [mockTransports.inMemory(logOutput)]
         });
 
-        const logger = winston.createLogger({
-          transports: [
-            mockTransport.createMockTransport(assertFn)
-          ]
+        logger1.info("some message");
+        logger1.warn("I'm a test");
+        logger2.log({level: 'info', message: "some message"});
+        logger2.log({level: 'warn', message: "I'm a test"});
+
+        assume(levelOutput).eqls(logOutput);
+        assume(expectedOutput).eqls(levelOutput);
+        assume(expectedOutput).eqls(logOutput);
+      });
+
+      it('should include default metadata defined on the logger instance', () => {
+        const expectedOutput = [
+          {message: "some message", level: "info", defaultMeta: true},
+        ];
+
+        const logger1 = winston.createLogger({
+          transports: [mockTransports.inMemory(levelOutput)],
+          defaultMeta: {defaultMeta: true}
```

Comment:
I don't like that the value of defaultMeta is an object with key defaultMeta 😂  Any chance we can make the object something less... meta? 😅

Reply:
Hah yea I can change it. Meant it more as a `isDefaultMeta` 😅